### DISCUSSION TITLE
How would you conditionally send (or not send) static files to the client with middleware?

### DISCUSSION THREAD

**Discussion by SamRoehrich** [vercel/next.js#39000] [Help]
Hey.
Consider that you have an application with an admin and non-admin interface and you do not want to send the admin code to the client for users who are not an admin, how would you do that? I watched the most recent video from Vercel and in the Q&A @leerob [mentioned that you could do this](https://youtu.be/j7rPSS9Ovsw?t=2270). Curious if anyone has a repo demonstrating this feature / could help me understand how to go about implementing this myself.

**Comment by wwwdepot** [comment]
Example: https://github.com/vercel/examples/tree/main/edge-functions/jwt-authentication
